target/cheri-common/cheri_utils.h

```diff
@@ -202,30 +233,41 @@ static inline bool cap_is_unsealed(const cap_register_t *c)
 {
     // TODO: how should we treat the other reserved types? as sealed?
     // TODO: what about untagged capabilities with out-of-range otypes?
+#ifdef TARGET_AARCH64
+    return cap_get_otype_unsigned(c) == CAP_OTYPE_UNSEALED;
+#else
     _Static_assert(CAP_MAX_REPRESENTABLE_OTYPE == CAP_OTYPE_UNSEALED, "");
     if (c->cr_tag) {
         cheri_debug_assert(cap_get_otype_unsigned(c) <=
                            CAP_MAX_REPRESENTABLE_OTYPE);
     }
     return cap_get_otype_unsigned(c) >= CAP_OTYPE_UNSEALED;
+#endif
 }
 
 static inline void cap_set_sealed(cap_register_t *c, uint32_t type)
 {
+#ifdef TARGET_AARCH64
+    type &= CAP_CC(FIELD_OTYPE_MASK_NOT_SHIFTED);
```

Comment:
When is this needed?

Reply:
This might be another case where mips/riscv faulted if type was too large, but morello just masked it. I think I stopped using the helper, but we should still maintain correctness for the helper as it is reasonable to call for morello, and should not break when it does.

Reply:
Ok, I read the spec, and Morello will truncate+tag clear in that case. That needs handling here if you want the helper to be correct, otherwise don't do anything here, having a half-baked implementation is the worst option.